virtual_ecosystem/models/animal/animal_cohorts.py

```diff
@@ -547,40 +642,46 @@ def calculate_consumed_mass_predation(
 
     def delta_mass_predation(
         self,
-        animal_list: Sequence[AnimalCohort],
-        excrement_pool: DecayPool,
-        carcass_pool: CarcassPool,
+        animal_list: list[AnimalCohort],
+        excrement_pools: list[ExcrementPool],
+        carcass_pools: dict[int, set[CarcassPool]],
```

Comment:
OK - I think I see why the cell_id keyed dicts here. I think the issue is that animals can only eat an animal and leave its carcass in overlapping territory cells, but defecation can occur anywhere. So you have to retain the `cell_id` on the carcass pools, so that `get_eaten` can drop corpses in the right locations.

Something that might make it simpler: 
* `AnimalCohort` has an attribute `excrement_pools` that is updated when territory changes. 
* `delta_mass_predation` works on a single prey item at a time. So something like:

```python
for prey in animal_list:
    self.delta_mass_predate(prey)

def delta_mass_predate(self, prey: AnimalCohort):
    intersection = self.territory & prey.territory  # assuming sets
    etc.
```

That seems cleaner than having to do a bunch of repeated indexing at multiple levels? I think we'd need to add the complete set of carcass pools  to each instance?